server/src/test/java/io/crate/planner/operators/TableFunctionTest.java

```diff
@@ -0,0 +1,83 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.operators;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Test;
+
+import io.crate.analyze.WhereClause;
+import io.crate.analyze.relations.TableFunctionRelation;
+import io.crate.expression.symbol.Symbol;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+
+public class TableFunctionTest extends CrateDummyClusterServiceUnitTest {
+
+    @Test
+    public void test_column_pruning_with_empty_outputs() throws IOException {
+        SQLExecutor e = SQLExecutor.of(clusterService).addTable("create table tbl (a int, b int)");
+
+        Symbol a = e.asSymbol("a");
+        Symbol b = e.asSymbol("b");
+
+        TableFunctionRelation tableFunctionRelation = mock(TableFunctionRelation.class);
+        LogicalPlan tableFunction = TableFunction.create(tableFunctionRelation, List.of(), WhereClause.MATCH_ALL);
+        assertThat(tableFunction.outputs()).isEmpty();
+        var prunedTableFunction = tableFunction.pruneOutputsExcept(List.of(a, b));
+        assertThat(prunedTableFunction.outputs()).containsExactly(a, b);
+    }
+
+    @Test
+    public void test_column_pruning_with_outputs() throws IOException {
+        SQLExecutor e = SQLExecutor.of(clusterService).addTable("create table tbl (a int, b int)");
+
+        Symbol a = e.asSymbol("a");
+        Symbol b = e.asSymbol("b");
+
+        TableFunctionRelation tableFunctionRelation = mock(TableFunctionRelation.class);
+        LogicalPlan tableFunction = TableFunction.create(tableFunctionRelation, List.of(a, b), WhereClause.MATCH_ALL);
+        assertThat(tableFunction.outputs()).containsExactly(a, b);
+        var prunedTableFunction = tableFunction.pruneOutputsExcept(List.of(a));
+        assertThat(prunedTableFunction.outputs()).containsExactly(a);
+    }
+
+    @Test
+    public void test_column_pruning_with_more_outputs() throws IOException {
+        SQLExecutor e = SQLExecutor.of(clusterService).addTable("create table tbl (a int, b int, c int)");
+
+        Symbol a = e.asSymbol("a");
+        Symbol b = e.asSymbol("b");
+        Symbol c = e.asSymbol("c");
+
+        TableFunctionRelation tableFunctionRelation = mock(TableFunctionRelation.class);
+        LogicalPlan tableFunction = TableFunction.create(tableFunctionRelation, List.of(a, b, c), WhereClause.MATCH_ALL);
+        assertThat(tableFunction.outputs()).containsExactly(a, b, c);
+        var prunedTableFunction = tableFunction.pruneOutputsExcept(List.of(a, b));
```

Comment:
can we do something more complex, to be maybe a bit more future proof:
`List .of(c, b)` for example.